agentos/core.py

```diff
@@ -45,14 +52,58 @@ class Agent(MemberInitializer):
     learning, use of models, state updates, etc.
     """
 
-    def learn(self):
-        """Does one iteration of training"""
-        pass
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.curr_obs = None
+        self._should_reset = True
+
+    def step(self):
+        """Takes one action within the environment"""
+        if self._should_reset:
+            self.curr_obs = self.environment.reset()
+            self._should_reset = False
+            self.dataset.add(None, None, self.curr_obs, None, None, {})
+        action = self.policy.decide(
+            self.curr_obs, self.environment.valid_actions
+        )
+        prev_obs = self.curr_obs
+        self.curr_obs, reward, done, info = self.environment.step(action)
+        self.dataset.add(prev_obs, action, self.curr_obs, reward, done, info)
+        if done:
+            self._should_reset = True
+        return prev_obs, action, self.curr_obs, reward, done, info
+
+    def rollout(self, should_learn):
```

Comment:
Good point.  Acme actually handles this by calling their `train()` function (called [`Agent.update()`](https://github.com/deepmind/acme/blob/master/acme/environment_loop.py#L100) in their case) after *every* step in the environment and then Trainer is responsible for [deciding if/how much it will actually train during that particular call](https://github.com/deepmind/acme/blob/master/acme/agents/agent.py#L80).

I think I like the Acme behavior and will update this function to mimic it.  This would make it a convention that the Trainer gets a chance to train after every step in the environment (and it can decide to take that chance or not).  

Let me know how you feel about this design choice; happy to consider alternatives as well.